frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -579,6 +600,37 @@ common_settings.document_metadata_location = {
         },
         genMetadataFolderMenuItem("doc"),
         genMetadataFolderMenuItem("dir"),
+        genMetadataFolderMenuItem("hash"),
+        { -- hash-based metadata count / TextViewer
+            text_func = function()
+                local hash_text = _("Show documents with hash-based metadata")
+                local no_hash_text = _("No documents with hash-based metadata")
+                if G_reader_settings:readSetting("document_metadata_folder") == "hash" then
+                    if DocSettings.isHashLocationEnabled() then return hash_text end
+                    -- displays if the hash location setting is selected but no hash-based metadata is found
+                    return no_hash_text
+                end
+                if DocSettings.isHashLocationEnabled() then return  "⚠ "..hash_text end
+                return no_hash_text
+            end,
+            keep_menu_open = true,
+            enabled_func = function()
+                return DocSettings.isHashLocationEnabled()
+            end,
+            callback = function()
+                local hash_file_infos = DocSettings.getHashDirSdrInfos()
+                local book_info_list = ""
+                for i, file_info in ipairs(hash_file_infos) do
+                    book_info_list = book_info_list.."\n"..i..". "..file_info
+                end
+                UIManager:show(require("ui/widget/textviewer"):new{
+                    title = T(_("%1 documents with hash-based metadata"), #hash_file_infos),
```

Comment:
`N_` required.

Reply:
What is this? I'm actually not sure why this has a `T` either - I think the `_()` function is for translatable strings. Should it read `T(N_("%1 documents with hash-based metadata"), #hash_file_infos)`?

Reply:
https://github.com/koreader/koreader/blob/5b5b4d9ebc970ee80d49adf66021d71af15b317c/frontend/gettext.lua#L293-L321